service_container/service_decoration.rst

```diff
@@ -313,4 +313,119 @@ The generated code will be the following::
+option will allow you to handle the lack of decorated service.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        Foo: ~
+
+        Bar:
+            public: false
+            decorates: Foo
+            decoration_on_invalid: ignore
+            arguments: ['@Bar.inner']
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="Foo"/>
+
+                <service id="Bar" decorates="Foo" decoration-on-invalid="ignore">
+                    <argument type="service" id="Bar.inner"/>
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+        use Symfony\Component\DependencyInjection\ContainerInterface;
+
+        return function(ContainerConfigurator $configurator) {
+            $services = $configurator->services();
+
+            $services->set(Foo::class);
+
+            $services->set(Bar::class)
+                ->decorate(Foo::class, null, 0, ContainerInterface::IGNORE_ON_INVALID_REFERENCE)
+                ->args([ref(Bar::class.'.inner')]);
+
+            $services->set(Baz::class)
+                ->private()
+                ->decorate(Foo::class, null, 1)
+                ->args([ref(Baz::class.'.inner')]);
+        };
+
+Three different behaviors are available:
+
+* ``exception``: The default behavior. When used, a ``ServiceNotFoundException``
+  will be thrown telling that decorator's dependency is missing.
+* ``ignore``: Using that behavior, the container will remove either decorator
+  and decorated services.
+* ``null``: When used, the container will keep the decorator
+  service and will set the decorated one to `null`.
+
+.. tip::
+
+    When using ``null``, you may update the decorator's constructor in
+    order to make decorated dependency nullable. In this way, you will be able to
+    check the presence of decorated service in other methods.
+
+    .. code-block:: yaml
+
+        App\Service\DecoratorService:
+            decorates: Acme\OptionalBundle\Service\OptionalService
+            decoration_on_invalid: null
+            arguments: ['@App\Service\DecoratorService.inner']
+
+    ::
+
+        namespace App\Service;
+
+        use Acme\OptionalBundle\Service\OptionalService;
+
+        class DecoratorService
+        {
+            private $decorated;
+
+            - public function __construct(OptionalService $decorated)
+            + public function __construct(?OptionalService $decorated)
```

Comment:
I am not a fan of the `-` and `+` here, at least you need to use `.. code-block: diff` to get correct highlighting.
Why not just show up the way to go? I mean only the `+` case

Reply:
Yes you're fully right ! Thanks